test/rules/use_build_context_synchronously_test.dart

```diff
@@ -353,8 +351,123 @@ void foo(BuildContext context, int i) async {
     expect(forElement.asyncStateFor(reference), AsyncState.mountedCheck);
   }
 
+  test_forStatementWithDeclaration_referenceAfter_awaitInPartCondition() async {
```

Comment:
Exhaustive testing 👍

Reply:
Haha yeah, kind of tiresome, but... I guess since these are all encoded in the visitor, its good to cover them.